### PR TITLE
Manage properly the CLI error and implement unit test for the CLI

### PR DESCRIPTION
**Pull request by Nexucis**
This PR is changing and introducing different things:
1. First it changes the way to manage the error in the CLI. Basically all errors are reported to the root CMD which is able to handle and print the error in the correct file descriptor.
2. It introduces a way to inject an `io.writer` in the cmd so we can intercept what is printed on stdout / stderr and verify it is the correct message (for testing purpose)
3. It provides also a fake http client that should be used when you want to mock the connection to the API. It's easier to use than to use an actual mock system.
4. It also provides an interface that each command must implement.

### CODE REVIEW

File: internal/cli/utils/test/test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/perses/perses/pkg/client/api"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+)
+
+func JSONMarshalStrict(obj interface{}) []byte {
+	if data, err := json.Marshal(obj); err != nil {
+		panic(err)
+	} else {
```
Comment (juliusv) [inline internal/cli/utils/test/test.go:31]: Omit the else when possible after error handling, see https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow.

File: internal/cli/utils/test/test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		return data
+	}
+}
+
+func YamlMarshalStrict(obj interface{}) []byte {
```
Comment (juliusv) [inline internal/cli/utils/test/test.go:36]: `Yaml` -> `YAML` (https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

File: pkg/client/fake_api/fake_v1/folder.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	modelV1 "github.com/perses/perses/pkg/model/api/v1"
+)
+
+func FolderList(project string, prefix string) []*modelV1.Folder {
+	initialList := []*modelV1.Folder{
+		{
+			Kind: modelV1.KindFolder,
+			Metadata: modelV1.ProjectMetadata{
+				Metadata: modelV1.Metadata{
+					Name: "FF15",
```
Comment (juliusv) [inline pkg/client/fake_api/fake_v1/folder.go:29]: Final Fantasy 15?

**Review by juliusv — COMMENTED** [review]
:+1: besides nits

File: pkg/client/fake_api/fake_v1/folder.go (hunk 3 above)
Comment (Nexucis) [reply pkg/client/fake_api/fake_v1/folder.go:29]: haha yes :).  I was searching some inspirations ^^

**Comment by Nexucis** [comment]
thanks @juliusv !

**Pull request merged**